lib/acts_as_tree.rb

```diff
@@ -141,7 +141,7 @@ def leaves
         class_eval <<-EOV
           def self.leaves
             internal_ids = select(:#{configuration[:foreign_key]}).where(arel_table[:#{configuration[:foreign_key]}].not_eq(nil))
-            where("id NOT IN (\#{internal_ids.to_sql})").default_tree_order
+            where("#{configuration[:primary_key]} NOT IN (\#{internal_ids.to_sql})").default_tree_order
```

Comment:
This must use `quote_column_name` or should be changed to use arel. There is no guarantee, that the value given to `:primary_key` is not a reserved keyword or contains characters that require quoting.